Init version of 40 msprime installs across 18 distros

castedo opened the pull request:
Types of tskit installs via
  - miniconda
  - conda from distro system package
  - pip
  - pip --pre

This PR is the functionality described in #2 ~~but only for tskit installs~~.

Closes #2

castedo committed:
Init version of 40 tskit installs across 18 distros

  Types of tskit installs via
  - miniconda
  - conda from distro system package
  - pip
  - pip --pre

castedo committed:
test msprime pip install on opensuse; refactor with _taxa and _common

castedo committed:
test msprime on yum distros; refactor away yum-pip* tests

castedo committed:
test msprime on apt distros; refactor away apt-pip* tests

castedo committed:
refactor all remaining tests to test msprime

castedo commented:
I've upgraded and tested all scripts to test msprime install instead of just tskit. Also refactored to reuse common test stages.

jeromekelleher commented:
Wow, this is mega-thorough, thanks @castedo. I don't see any issues, but I'm not a container-wizard.

What's your vision for keeping this up-to-date? Is it something we rerun periodically, and git-commit the pass/fail statuses?

castedo commented:
> What's your vision for keeping this up-to-date? Is it something we rerun periodically, and git-commit the pass/fail statuses?

@jeromekelleher The tests seem likely enough to succeed that it does not seem worth it to setup it up to run automatically on a regular basis. Seems to me more of a useful exploratory occasional test to be kicked off manually from time to time. Maybe a bit like verification.py.

To be honest, I was expecting miniconda to NOT perform so well and was curious when it would fail. But miniconda performed great! Looks like a great solution for a very relevant set of scenarios.

@benjeffery and I have chatted on Slack briefly. When binary wheels are up on PyPI I can kick off the script again and confirm install passed for the ancient distros that currently fail. And I can git-commit to PASS status.

castedo commented:
It is worth mentioning that this test does not test with really old Linux kernels. The script is not spinning up VMs, just running containers. So it's really testing whether there are unexpected dependencies or unhandled variation across many diff distros. So, in theory, something might not work with ancient Linux kernels, but everything passes in this script when running on a relatively new Linux kernel (like I am).

jeromekelleher commented:
Sounds good @castedo.

castedo committed:
add 8 most disto-tests; mainly min-pip

castedo committed:
Fix bug where min-pip was not catching issues

  - the min-pip test script for yum was not actually
    installing system packages for python modules

jeromekelleher commented:
OK, merging because of request over in #2 (usually better to ping on the PR thread, though, to avoid confusion).
